lib/Common/Memory/CustomHeap.cpp

```diff
@@ -294,7 +295,11 @@ BOOL Heap<TAlloc, TPreReservedAlloc>::ProtectAllocationWithExecuteReadWrite(Allo
     }
     else
     {
+        #if defined(__APPLE__) && defined(_M_ARM64)
+        protectFlags = PAGE_READWRITE; // PAGE_EXECUTE_READWRITE banned on Apple Silicon
```

Comment:
Side note: this would be trouble for JIT in the future

Reply:
I was worried about it but I'm not sure if it will be a problem, I was able to get some run time produced code to run by setting a page to READWRITE, writing it and then switching to EXECUTE_READ before execution.